src/app/models/jobs-response.model.ts

```diff
@@ -0,0 +1,90 @@
+export class JobsResponse {
+  readonly flowStatus: string;
+  readonly flowStatusMessage: string;
+  readonly apiResponse: any;
+  readonly rawJobsData: any;
+  readonly jobs: JobData[];
+
+  constructor(response: any) {
+    try {
+      if (response.flowStatus === 'SUCCESS' || response.flowStatus === 'FAILURE') {
+        this.flowStatus = response.flowStatus;
+        this.flowStatusMessage = response.flowStatusMessage;
+      }
+      else {
+        this.flowStatus = 'FAILURE';
+        this.flowStatusMessage = 'Unable to call API';
+      }
+      if (response.flowStatus === 'SUCCESS') {
+        // save data from response
+        this.apiResponse = response;
+        this.rawJobsData = response.jobData;
+        this.jobs = response.jobData.jobs;
+        for (let i = 0; i < this.jobs.length; i++) {
+          const jobType = this.jobs[i].jobType;
+          this.jobs[i].jobTypeColor = this.parseJobTypeColor(jobType);
+        }
+      }
+    }
+    catch (e) {
+      this.flowStatus = 'FAILURE';
+      this.flowStatusMessage = 'Unable to parse API response';
+    }
+  }
+
+  private parseJobTypeColor(jobType: string): string {
+    return jobType === 'Install' ? '#0764a9' : jobType === 'Repair' ? '#563064' : jobType === 'Helper' ? '#050' : jobType === 'BSW' ? '#3a2204' : jobType === 'POTS' ? '#5e5c13' : 'black';
+  }
+}
+
+// IMPORTANT INFO:
+// The following interfaces are copied directly from the NestJS server ABCNest, and must match exactly
+// HOWEVER:
+// jobTypeColor: string is added to the JobData interface here, as that data is not calculated in the back end.
+export interface JobData {
+  accountNumber: number,
+  firstName: string,
+  lastName: string,
+  streetAddress: string,
+  city: City,
```

Comment:
refactor "city" to maybe location?

Reply:
I went ahead and did this. City is now refactored to Location. Also moved streetAddress variable from JobData into Location